src/Bicep.Core.UnitTests/Diagnostics/LinterRuleTests/UseResourceIdFunctionsRuleTests.cs

```diff
@@ -0,0 +1,1302 @@
+                  detector: {
+                    id: concat('someDetectorID')
+                  }
+                }
+                tags: {
+                  intagId: concat('some data')
+                }
+              }
+            }
+
+            resource service_backend 'Microsoft.ApiManagement/service/backends@2020-06-01-preview' = {
+              name: concat('service/backend')
+              properties: {
+                description: 'desc'
+                resourceId: 'https://management.azure.com${resourceId('Microsoft.Web/sites', 'web')}'
+                url: 'https://${reference(resourceId('Microsoft.Web/sites', 'web'), '2020-09-01').defaultHostName}'
+                protocol: 'http'
+              }
+            }",
+            new object[]
+            {
+                // pass
+            },
+            DisplayName = "Pass/exceptions-that-should-pass.json")]
+        //// from https://github.com/Azure/arm-ttk/blob/master/unit-tests/IDs-Should-Be-Derived-From-ResourceIDs/Pass/logic-app-that-contain-id-should-pass.json
+        [DataRow(
+        @"
+            resource mylogicapp 'Microsoft.Logic/workflows@2019-05-01' = {
+              name: 'mylogicapp'
+              properties: {
+                definition: {
+                  '$schema': 'https://schema.management.azure.com/providers/Microsoft.Logic/schemas/2016-06-01/workflowdefinition.json#'
+                  contentVersion: '1.0.0.0'
+                  triggers: {
+                    manual: {
+                      correlation: {
+                        clientTrackingId: concat('@{coalesce(triggerOutputs().headers?[\'id\'], guid())}')
+                      }
+                      emailId: concat('@body(\'Get_email_(V2)_-_Processing\')?[\'id\']')
+                    }
+                    somedataId: concat('@concat(parameters(\'firstName\'), parameters(\'lastName\'))')
+                  }
+                }
+              }
+            }",
+            new object[]
+            {
+                // pass
+            },
+            DisplayName = "logic-app-that-contain-id-should-pass.json")]
+        [DataRow(
+        @"
+            resource mylogicapp 'Microsoft.Logic/notWorkflows@2019-05-01' = {
+              name: 'mylogicapp'
+              properties: {
+                guid: 'abc/def'   // ignore
+                myGuid: 'abc/def' // ignore
+                myUuid: 'abc/def' // ignore
+                myUUID: 'abc/def' // ignore
+                guId: 'abc/def'   // fail
+              }
+            }",
+            new object[]
+            {
+                "[9:17] If property \"guId\" represents a resource ID, it must use a symbolic resource reference (preferred) or start with one of these functions: extensionResourceId, guid, if, reference, resourceId, subscription, subscriptionResourceId, tenantResourceId."
+            },
+            DisplayName = "ignore property names ending in 'guid'")]
```

Comment:
DisplayName doesn't match the error message here. Could you please update?

Reply:
That one case fails intentionally as commented in the bicep code above.  Added another comment.